### PR TITLE
Swag custom image build

### PR DESCRIPTION
**Pull request by ofersadan85**
2 Big changes here
## Added our own custom build to SWAG
There's a [new image in Docker hub](https://hub.docker.com/r/ohad24/encyclo-flower-swag) and a new workflow to update it

Reasons:
- Our current `deploy.sh` pulls the most updated docker images, including SWAG, this can result in breaking changes if the new updated SWAG image changes something we don't know about. This will be more stable, more maintainable, and reversible.
- Dependabot will be able to update the pinned upstream SWAG images and open a pull request, so when there's a version change we can choose if we want to apply it or not (`dependabot.yml` will be updated after merging)
- Will be easier to switch to something other than SWAG if we need to in the future (another variant of nginx)

## Rewrote nginx config files
They do basically the same, but:
- Remove the bloat that comes built-in in SWAG and focuses on our config
- This will be easier to switch to another nginx variant if we want to
- Ensures that production and development configs are the same, without maintaining two separate configs

# Important note
I needed to create the workflow on the main branch (otherwise I can't test it) and cherry-picked from it on this branch after some fixes, so if you want to rebase this I think it has to be done on main after the merge (unless you know how to do it better)

### CODE REVIEW

File: server/docker-compose.yml [hunk 1]
```diff
@@ -1,16 +1,13 @@
 version: "3.9"
 services:
   swag:
-    image: linuxserver/swag:latest
+    image: ohad24/encyclo-flower-swag:edge  # TODO: Switch to "latest"
```
Comment (ohad24) [inline server/docker-compose.yml:4]: why edge and not latests ?

File: swag/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Using pinned version number, it will be updated by dependabot
+FROM linuxserver/swag:1.25.0
```
Comment (ohad24) [inline swag/Dockerfile:2]: How depandbot will check if this version will update ?

**Comment by ohad24** [comment]
Looking good

File: server/docker-compose.yml (hunk 1 above)
Comment (ofersadan85) [reply server/docker-compose.yml:4]: Our swag gateway doesn't have a dev version so this is temporary, latest tag will be applied soon after some testing

File: swag/Dockerfile (hunk 2 above)
Comment (ofersadan85) [reply swag/Dockerfile:2]: Something like this, I'll add it soon


  ```
 ​version​: ​2 
 ​updates​: 
 ​  - ​package-ecosystem​: ​"​docker​" 
 ​    ​directory​: ​"​swag" 
 ​    ​schedule​: 
 ​      ​interval​: ​"​daily​"
```

Comment (ofersadan85) [reply swag/Dockerfile:2]: See [053f5c0](https://github.com/ohad24/encyclo-flower/pull/53/commits/053f5c074ad685a673c54c4f33a08bf02ad2f93e)

File: server/docker-compose.yml (hunk 1 above)
Comment (ohad24) [reply server/docker-compose.yml:4]: Opend https://github.com/ohad24/encyclo-flower/issues/55

File: swag/Dockerfile (hunk 2 above)
Comment (ohad24) [reply swag/Dockerfile:2]: Thanks

**Comment by ohad24** [comment]
Looking good. Please merge whenever you ready

**Pull request merged**